File: 수학(구현)/백준_1913_달팽이/달팽이.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args)throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int m = s.nextInt();

        int[][] arr= new int[1000][1000];
        int k = 1;
        int i = n/2;
        int j = n/2;
        int num = 1;

        arr[i][j] = num;
        while(k <= n){
            if(k == n){
                for(int r = 0; r< k -1; r++){
                    i--;
                    num++;
                    arr[i][j] = num;
                }
            }
            else if (k % 2 != 0)
            {
                for (int r = 0; r < k; r++)
                {
                    i--;
                    num++;
                    arr[i][j] = num;
                }

                for (int r = 0; r < k; r++)
                {
                    j++;
                    num++;
                    arr[i][j] = num;
                }
            }
            else
            {
                for (int r = 0; r < k; r++)
                {
                    i++;
                    num++;
                    arr[i][j] = num;
                }

                for (int r = 0; r < k; r++)
                {
                    j--;
                    num++;
                    arr[i][j] = num;
                }
            }

            k++;
        }
        int ans_i = 0;
        int ans_j = 0;
        for(int a= 0; a < n; a++){
            for(int b = 0; b < n; b++){
                bw.write(arr[a][b] + " ");
                if(arr[a][b] == m){
                    ans_i = a;
                    ans_j = b;
                }
            }
            bw.newLine();
        }
        bw.write((ans_i+1) + " " + (ans_j+1));
        bw.flush();
    }
}

File: 수학(구현)/백준_1913_달팽이/달팽이.kt
import java.util.*

fun main(args: Array<String>) {
    val sb = StringBuilder()
    val s = Scanner(System.`in`)
    val n = s.nextInt()
    val m = s.nextInt()

    var arr = Array(1000, {IntArray(1000, {0})})
    var k = 1
    var i = n/2
    var j = n/2
    var num = 1

    arr[i][j] = num
    while(k <= n){
        if(k == n){
            for(r in 0 until k-1){
                i--
                num++
                arr[i][j] = num
            }
        } else if (k % 2 != 0) {
            for (r in 0 until k) {
                i--
                num++
                arr[i][j] = num
            }
            for (r in 0 until k) {
                j++
                num++
                arr[i][j] = num
            }
        } else {
            for (r in 0 until k) {
                i++
                num++
                arr[i][j] = num
            }
            for (r in 0 until k) {
                j--
                num++
                arr[i][j] = num
            }
        }
        k++
    }

    var ans_i = 0
    var ans_j = 0
    for(a in 0 until n){
        for( b in 0 until n){
            sb.append("${arr[a][b]} ")
            if(arr[a][b] == m){
                ans_i = a
                ans_j = b
            }
        }
        sb.append("\n")
    }
    sb.append("${ans_i + 1} ${ans_j + 1}")
    println(sb)
}